heimdall2/templates/heimdall-statefulset.yaml

```diff
@@ -68,209 +72,326 @@ spec:
+            {{- if .Values.oidcClientSecret }}
             - name: OIDC_CLIENT_SECRET
               valueFrom:
                 secretKeyRef:
                   name: {{ include "heimdall.fullname" . }}
                   key: oidcClientSecret
+            {{- end }}
+            {{- if .Values.oidcExternalGroups }}
             - name: OIDC_EXTERNAL_GROUPS
               value: {{ .Values.oidcExternalGroups | quote }}
+            {{- end }}
+            {{- if .Values.ldapEnabled }}
             - name: LDAP_ENABLED
               value: {{ .Values.ldapEnabled | quote }}
+            {{- end }}
+            {{- if .Values.ldapHost }}
             - name: LDAP_HOST
               value: {{ .Values.ldapHost | quote }}
+            {{- end }}
+            {{- if .Values.ldapPort }}
             - name: LDAP_PORT
               value: {{ .Values.ldapPort | quote }}
+            {{- end }}
+            {{- if .Values.ldapBindDn }}
             - name: LDAP_BINDDN
               valueFrom:
                 secretKeyRef:
                   name: {{ include "heimdall.fullname" . }}
                   key: ldapBindDn
+            {{- end }}
+            {{- if .Values.ldapPassword }}
             - name: LDAP_PASSWORD
               valueFrom:
                 secretKeyRef:
                   name: {{ include "heimdall.fullname" . }}
                   key: ldapPassword
+            {{- end }}
+            {{- if .Values.ldapSearchBase }}
             - name: LDAP_SEARCHBASE
               value: {{ .Values.ldapSearchBase | quote }}
+            {{- end }}
+            {{- if .Values.ldapSearchFilter }}
             - name: LDAP_SEARCHFILTER
               value: {{ .Values.ldapSearchFilter | quote }}
+            {{- end }}
+            {{- if .Values.ldapNameField }}
             - name: LDAP_NAMEFIELD
               value: {{ .Values.ldapNameField | quote }}
+            {{- end }}
+            {{- if .Values.ldapMailField }}
             - name: LDAP_MAILFIELD
               value: {{ .Values.ldapMailField | quote }}
+            {{- end }}
+            {{- if .Values.ldapSsl }}
             - name: LDAP_SSL
               value: {{ .Values.ldapSsl | quote }}
+            {{- end }}
+            {{- if .Values.ldapSslInsecure }}
             - name: LDAP_SSL_INSECURE
               value: {{ .Values.ldapSslInsecure | quote }}
+            {{- end }}
+            {{- if .Values.ldapSslCa }}
             - name: LDAP_SSL_CA
               valueFrom:
                 secretKeyRef:
                   name: {{ include "heimdall.fullname" . }}
                   key: ldapSslCa
+            {{- end }}
           livenessProbe:
+            {{- with .Values.heimdall.livenessProbe }}
             httpGet:
-              path: /
-              port: 3000
-            initialDelaySeconds: 80
-            periodSeconds: 10
-            timeoutSeconds: 15
-            failureThreshold: 3
+              path: {{ .httpGet.path | default "/" }}
```

Comment:
Helm docs say "In an actual chart, all static default values should live in the values.yaml, and should not be repeated using the default command (otherwise they would be redundant)."  My read on this is that if .Values.heimdall.livenessProbe is set then you should assume that .httpGet.path, etc are all set so there is no need to use the default keyword.

Please remove all uses of the default keyword unless it's for the exception also stated in the docs for a computed value, i.e. have to concat two different values together or something